W3ChampionsStatisticService/PlayerProfiles/PlayersController.cs

```diff
@@ -40,49 +40,24 @@ public async Task<IActionResult> GlobalSearchPlayer(string search, string lastRe
     }
 
     [HttpGet("{battleTag}")]
-    public async Task<IActionResult> GetPlayer([FromRoute] string battleTag, [FromQuery] string authorization)
+    public async Task<IActionResult> GetPlayer([FromRoute] string battleTag)
     {
-        var player = await _playerRepository.LoadPlayerProfile(battleTag);
+        PlayerOverallStats player = await _playerRepository.LoadPlayerProfile(battleTag);
 
         if (player == null)
         {
-            if (authorization == null)
-            {
-                return NotFound($"Player {battleTag} not found.");
-            }
-            else
-            {
-                try
-                {
-                    _authenticationService.GetUserByToken(authorization, false);
-                }
-                catch (Exception)
-                {
-                    return Unauthorized("Sorry Hackerboi");
-                }
-                player = PlayerOverallStats.Create(battleTag);
-            }
+            return NotFound($"Player {battleTag} not found.");
         }
 
         // Akas are stored in cache - preferences for showing akas are stored in DB
-        var settings = await _personalSettingsRepository.Load(battleTag);
+        PersonalSetting settings = await _personalSettingsRepository.LoadOrCreate(battleTag);
         player.PlayerAkaData = await _playerAkaProvider.GetAkaDataByPreferencesAsync(battleTag, settings);
 
         await _playerRepository.UpsertPlayer(player);
```

Comment:
Why are we updating every time we get a player?

Reply:
The repository does nothing if the element hasn't actually changed, pretty sure.